server/core/region.go

```diff
@@ -165,18 +168,20 @@ func RegionFromHeartbeat(heartbeat *pdpb.RegionHeartbeatRequest, opts ...RegionC
 	return region
 }
 
-// CorrectApproximateSize correct approximate size by the previous size if here exists an reported RegionInfo.
-//
+// Inherit inherits the buckets and region size from the parent region.
+// correct approximate size and buckets by the previous size if here exists a reported RegionInfo.
 // See https://github.com/tikv/tikv/issues/11114
-func (r *RegionInfo) CorrectApproximateSize(origin *RegionInfo) {
-	if r.approximateSize != 0 {
-		return
+func (r *RegionInfo) Inherit(origin *RegionInfo) {
+	// regionSize should not be zero if region is not empty.
```

Comment:
I think we can check whether the `origin` is nil at the beginning of this function and return directly if it is.

Reply:
region size will be 1 if origin is nil and the size of the  new region  is zero.